clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -982,6 +983,12 @@ public void onFailure(RuntimeException e) {
                         }
                     }
                 }
+            } catch (AuthenticationException e) {
```

Comment:
I'm not sure this is the right place for this check. I think it would have to be in the `onFailure` hook above. A test case might clarify this.

Reply:
@hachikuji Since heartbeat thread is started only after a join request succeeds, this is perhaps not needed for this PR?

Reply:
Yes, if we are only concerned with authentication errors during initialization, then we can leave it for later. We should make sure to capture the gaps in another JIRA though.

Reply:
Created https://issues.apache.org/jira/browse/KAFKA-5945